gmpcompat.h

```diff
@@ -290,21 +290,16 @@ double flint_mpn_get_d (mp_srcptr ptr, mp_size_t size, mp_size_t sign, long exp)
 static __inline__
 void flint_mpz_set_si(mpz_ptr r, slong s)
 {
-   /* GMP 6.2 lazily performs allocation, deal with that if necessary
+    /* GMP 6.2 lazily performs allocation, deal with that if necessary
       (in older GMP versions, this code is simply never triggered) */
-   if (r->_mp_alloc == 0)
-   {
-      r->_mp_d = (mp_ptr) flint_malloc(sizeof(mp_limb_t));
-      r->_mp_alloc = 1;
-   }
+    if (r->_mp_alloc == 0)
+    {
+        r->_mp_d = (mp_ptr) flint_malloc(sizeof(mp_limb_t));
+        r->_mp_alloc = 1;
+    }
 
-   if (s < 0) {
-      r->_mp_size = -1;
-      r->_mp_d[0] = -s;
-   } else {
-      r->_mp_size = s != 0;
-      r->_mp_d[0] = s;
-   }
+    r->_mp_size = FLINT_SGN(s);
```

Comment:
Won't this slow it down? Previously it had to do one test for s < 0, now it has to do 2. Or is it a bug?

Reply:
I thought it was faster, so perhaps I changed something else when I tried this. Let's keep it as it is. Sorry for this.